samples/wifi/raw_tx_packet/src/main.c

```diff
@@ -198,15 +238,44 @@ static void wifi_send_raw_tx_packets(void)
 
 	buf_length = sizeof(struct raw_tx_pkt_header) + sizeof(test_beacon_frame);
 	memcpy(test_frame, &packet, sizeof(struct raw_tx_pkt_header));
-	memcpy(test_frame + sizeof(struct raw_tx_pkt_header),
+
+	if (num_pkts == 1) {
+		memcpy(test_frame + sizeof(struct raw_tx_pkt_header),
 				&test_beacon_frame, sizeof(test_beacon_frame));
 
-	ret = wifi_send_raw_tx_pkt(sockfd, test_frame, buf_length, &sa);
-	if (ret < 0) {
-		LOG_ERR("Unable to send beacon frame: %s", strerror(errno));
-		close(sockfd);
-		free(test_frame);
-		return;
+		ret = wifi_send_raw_tx_pkt(sockfd, test_frame, buf_length, &sa);
+		if (ret < 0) {
+			LOG_ERR("Unable to send beacon frame: %s", strerror(errno));
+			close(sockfd);
+			free(test_frame);
+			return;
+		}
+	} else {
+		while (1) {
```

Comment:
if you convert the code to use a for loop you can re-use the same code for both modes:

```
unsigned int num_pkts;

num_pkts = MAX_UINT if CONTINUOUS else num_pkts
for (i = 0; i < num_pkts; i++);
```
Of course this limits to MAX_UINT packets, but I guess that should be fine.